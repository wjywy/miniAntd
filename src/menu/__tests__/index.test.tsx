import { render } from '@testing-library/react';
import React from 'react';
import Menu from '../index';
import type { MenuProps } from '../index';
type MenuItem = Required<MenuProps>['items'][number];

function getItem(
  label: React.ReactNode,
  key: React.Key,
  icon?: React.ReactNode,
  children?: MenuItem[],
  type?: 'group',
): MenuItem {
  return {
    key,
    icon,
    children,
    label,
    type,
  } as MenuItem;
}

const items: MenuProps['items'] = [
  getItem('组件1', 'sub1', null, [
    getItem('子组件组1', 'g1', null, [getItem('组件项1', '1'), getItem('组件项2', '2')], 'group'),
    getItem('子组件组2', 'g2', null, [getItem('组件项3', '3'), getItem('组件项4', '4')], 'group'),
  ]),

  getItem('组件2', 'sub2', null, [
    getItem('组件项5', '5'),
    getItem('组件项6', '6'),
    getItem('子组件3', 'sub3', null, [getItem('组件项7', '7')]),
  ]),

  getItem('组件组', 'grp', null, [getItem('组件项8', '8'), getItem('组件项9', '9')], 'group'),
];

const { SubMenu } = Menu;

describe('Menu', () => {

  it('should render correctly', () => {
    const { getByTestId } = render(    
    <Menu
      data-testid="menu"
      style={{ width: 256 }}
      defaultSelectedKeys={['1']}
      defaultOpenKeys={['sub1']}
      mode="inline"
      items={items}
    />);
    expect(getByTestId('menu')).toMatchSnapshot();
  });

  it('should accept defaultOpenKeys in mode inline', () => {
    const { container } = render(
      <Menu defaultOpenKeys={['1']} mode="inline">
        <SubMenu key="1" title="submenu1">
          <Menu.Item key="submenu1">Option 1</Menu.Item>
          <Menu.Item key="submenu2">Option 2</Menu.Item>
        </SubMenu>
        <Menu.Item key="2">menu2</Menu.Item>
      </Menu>,
    );

    expect(
      container.querySelector('.cheesi-menu-submenu-open')?.querySelector('.cheesi-menu-submenu-title')
        ?.textContent,
    ).toEqual('submenu1');
  });

  it('If has select nested submenu item ,the menu items on the grandfather level should be highlight', () => {
    const { container } = render(
      <Menu defaultSelectedKeys={['1-3-2']} mode="vertical">
        <SubMenu key="1" title="submenu1">
          <Menu.Item key="1-1">Option 1</Menu.Item>
          <Menu.Item key="1-2">Option 2</Menu.Item>
          <SubMenu key="1-3" title="submenu1-3">
            <Menu.Item key="1-3-1">Option 3</Menu.Item>
            <Menu.Item key="1-3-2">Option 4</Menu.Item>
          </SubMenu>
        </SubMenu>
        <Menu.Item key="2">menu2</Menu.Item>
      </Menu>,
    );
    expect(container.querySelectorAll('li.cheesi-menu-submenu-selected').length).toBe(1);
  });
});
