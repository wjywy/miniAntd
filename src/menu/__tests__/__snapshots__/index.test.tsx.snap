// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `
<ul
  class="cheesi-menu cheesi-menu-root cheesi-menu-inline cheesi-menu-light"
  data-menu-list="true"
  data-testid="menu"
  dir="ltr"
  role="menu"
  style="width: 256px;"
  tabindex="0"
>
  <li
    class="cheesi-menu-submenu cheesi-menu-submenu-inline cheesi-menu-submenu-open cheesi-menu-submenu-selected"
    role="none"
  >
    <div
      aria-controls="rc-menu-uuid-94588-1-sub1-popup"
      aria-expanded="true"
      aria-haspopup="true"
      class="cheesi-menu-submenu-title"
      data-menu-id="rc-menu-uuid-94588-1-sub1"
      role="menuitem"
      style="padding-left: 24px;"
      tabindex="-1"
    >
      <span
        class="cheesi-menu-title-content"
      >
        组件1
      </span>
      <i
        class="cheesi-menu-submenu-arrow"
      />
    </div>
    <ul
      class="cheesi-menu cheesi-menu-sub cheesi-menu-inline"
      data-menu-list="true"
      id="rc-menu-uuid-94588-1-sub1-popup"
      role="menu"
    >
      <li
        class="cheesi-menu-item-group"
        role="presentation"
      >
        <div
          class="cheesi-menu-item-group-title"
          role="presentation"
          title="子组件组1"
        >
          子组件组1
        </div>
        <ul
          class="cheesi-menu-item-group-list"
          role="group"
        >
          <li
            class="cheesi-menu-item cheesi-menu-item-selected cheesi-menu-item-only-child"
            data-menu-id="rc-menu-uuid-94588-1-1"
            role="menuitem"
            style="padding-left: 48px;"
            tabindex="-1"
          >
            <span
              class="cheesi-menu-title-content"
            >
              组件项1
            </span>
          </li>
          <li
            class="cheesi-menu-item cheesi-menu-item-only-child"
            data-menu-id="rc-menu-uuid-94588-1-2"
            role="menuitem"
            style="padding-left: 48px;"
            tabindex="-1"
          >
            <span
              class="cheesi-menu-title-content"
            >
              组件项2
            </span>
          </li>
        </ul>
      </li>
      <li
        class="cheesi-menu-item-group"
        role="presentation"
      >
        <div
          class="cheesi-menu-item-group-title"
          role="presentation"
          title="子组件组2"
        >
          子组件组2
        </div>
        <ul
          class="cheesi-menu-item-group-list"
          role="group"
        >
          <li
            class="cheesi-menu-item cheesi-menu-item-only-child"
            data-menu-id="rc-menu-uuid-94588-1-3"
            role="menuitem"
            style="padding-left: 48px;"
            tabindex="-1"
          >
            <span
              class="cheesi-menu-title-content"
            >
              组件项3
            </span>
          </li>
          <li
            class="cheesi-menu-item cheesi-menu-item-only-child"
            data-menu-id="rc-menu-uuid-94588-1-4"
            role="menuitem"
            style="padding-left: 48px;"
            tabindex="-1"
          >
            <span
              class="cheesi-menu-title-content"
            >
              组件项4
            </span>
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    class="cheesi-menu-submenu cheesi-menu-submenu-inline"
    role="none"
  >
    <div
      aria-controls="rc-menu-uuid-94588-1-sub2-popup"
      aria-expanded="false"
      aria-haspopup="true"
      class="cheesi-menu-submenu-title"
      data-menu-id="rc-menu-uuid-94588-1-sub2"
      role="menuitem"
      style="padding-left: 24px;"
      tabindex="-1"
    >
      <span
        class="cheesi-menu-title-content"
      >
        组件2
      </span>
      <i
        class="cheesi-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    class="cheesi-menu-item-group"
    role="presentation"
  >
    <div
      class="cheesi-menu-item-group-title"
      role="presentation"
      title="组件组"
    >
      组件组
    </div>
    <ul
      class="cheesi-menu-item-group-list"
      role="group"
    >
      <li
        class="cheesi-menu-item cheesi-menu-item-only-child"
        data-menu-id="rc-menu-uuid-94588-1-8"
        role="menuitem"
        style="padding-left: 24px;"
        tabindex="-1"
      >
        <span
          class="cheesi-menu-title-content"
        >
          组件项8
        </span>
      </li>
      <li
        class="cheesi-menu-item cheesi-menu-item-only-child"
        data-menu-id="rc-menu-uuid-94588-1-9"
        role="menuitem"
        style="padding-left: 24px;"
        tabindex="-1"
      >
        <span
          class="cheesi-menu-title-content"
        >
          组件项9
        </span>
      </li>
    </ul>
  </li>
</ul>
`;
